### ISSUE TITLE
aws_iam_instance_profile doesn't export create_date

### ISSUE THREAD

**Issue by JousP** [terraform-providers/terraform-provider-aws#3527]
### Terraform Version
Terraform v0.11.3

### Affected Resource(s)
- aws_iam_instance_profile

### Terraform Configuration Files
```hcl
# Generates an IAM policy document
data "aws_iam_policy_document" "iam_assumeRole_generic" {
  statement {
    actions               = [ "sts:AssumeRole" ]
    principals {
      type                = "Service"
      identifiers         = [ "ec2.amazonaws.com" ]
    }
    effect                = "Allow"
  }
}

# Create the IAM role
resource "aws_iam_role" "role" {
  name                    = "ec2_AssumeRole"
  assume_role_policy      = "${data.aws_iam_policy_document.iam_assumeRole_generic.json}"
  description             = "Role for EC2 instance profile"
  path                    = "/"
}

# Create the instance profile 
resource "aws_iam_instance_profile" "main" {
  name                      = "ec2Profile"
  path                      = "/"
  role                      = "${aws_iam_role.role.name}"
}

## Instance profile outputs
output "id" {
  description = "The instance profile's ID."
  value       = "${aws_iam_instance_profile.main.id}"
}

output "arn" {
  description = "The ARN assigned by AWS to the instance profile."
  value       = "${aws_iam_instance_profile.main.arn}"
}

output "create_date" {
  description = "The creation timestamp of the instance profile."
  value       = "${aws_iam_instance_profile.main.create_date}"
}

output "name" {
  description = "The instance profile's name."
  value       = "${aws_iam_instance_profile.main.name}"
}

output "path" {
  description = "The path of the instance profile in IAM."
  value       = "${aws_iam_instance_profile.main.path}"
}

output "role" {
  description = "The role associated to the instance profile."
  value       = "${aws_iam_instance_profile.main.role}"
}

output "unique_id" {
  description = "The unique ID assigned by AWS."
  value       = "${aws_iam_instance_profile.main.unique_id}"
}
```

### Expected Behavior
Deploys an aws_iam_role and an aws_iam_instance_profile and output the aws_iam_instance_profile attributes

### Actual Behavior
`terraform apply`  end with following error : 
```
Error: Error applying plan:

1 error(s) occurred:

* output.create_date: Resource 'aws_iam_instance_profile.main' does not have attribute 'create_date' for variable 'aws_iam_instance_profile.main.create_date'

```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan -out deploy.out`
2. `terraform apply deploy.out`

**Comment by bootswithdefer** [comment]
Confirmed the resource doesn't provide create_date, but the data source does.  As a workaround you can do:
```
data "aws_iam_instance_profile" "example" {
  name = "example"
}

output "create_date" {
  description = "The creation timestamp of the instance profile."
  value       = "${data.aws_iam_instance_profile.example.create_date}"
}
```

**Comment by bflad** [comment]
The `aws_iam_instance_profile` resource `create_date` attribute will be released in version 1.25.0 of the AWS provider, likely middle of next week.

**Comment by bflad** [comment]
This has been released in [version 1.25.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1250-june-27-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
